<a href="https://colab.research.google.com/github/lwerth/escuelas_crimen_cdmx/blob/master/cdmx_escuelas_merged.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import numpy as np, scipy as sp, pymc3 as pm, matplotlib.pyplot as plt, pandas as pd, theano, theano.tensor as tt
from scipy import stats
import seaborn as sb
from sklearn.preprocessing import OrdinalEncoder, StandardScaler

from keras.preprocessing.text import Tokenizer,  text_to_word_sequence
import re
import nltk
from nltk import tokenize
import seaborn as sns
import json

%matplotlib inline
import matplotlib.pyplot as plt
from matplotlib.pyplot import *

import theano

In [43]:
#import fidegar files

fidegar = pd.read_excel('/content/drive/My Drive/Projects/CDMX Schools/fidegar.xlsx')
fidegar_idx = OrdinalEncoder(dtype = np.int64).fit_transform(fidegar[['NIVEL']])

fidegar['nivel'] = fidegar['NIVEL']
fidegar['nombre'] = fidegar['NOMBRE']
fidegar_idx
n_niveles_fidegar = len(fidegar.nivel.unique())
n_niveles_fidegar

fidegar.nivel.unique()
fidegar

NIVEL  ...                           nombre
0       PRIMARIA  ...                 JOSE INES NOVELO
1       PRIMARIA  ...       MAESTRO ENRIQUE COCA PEREZ
2     PREESCOLAR  ...           RICARDO ZAMORA BUITRON
3       PRIMARIA  ...                    EFRAIN HUERTA
4     PREESCOLAR  ...               REPUBLICA DE KENIA
...          ...  ...                              ...
2691  PREESCOLAR  ...                           PUNTLI
2692  SECUNDARIA  ...        GUILLERMO PRIETO PRADILLO
2693  PREESCOLAR  ...  PROFR. ISAIAS MORONES ESCAMILLA
2694    PRIMARIA  ...  PROFR. CARITINO MALDONADO PEREZ
2695  PREESCOLAR  ...       PROFR. LUCAS ORTIZ BENITEZ

[2696 rows x 11 columns]

In [60]:
# read in and look at gob data

#parse text in delegacion, delete the word "delegacion" in data, alphabetize, create index, and join!

publicas = pd.read_excel('/content/drive/My Drive/Projects/CDMX Schools/escuelas-publicas.xlsx')
pub_idx = OrdinalEncoder(dtype = np.int64).fit_transform(publicas[['nivel']])

n_niveles_pub = len(publicas.nivel.unique())
n_niveles_pub

n_delegacion_pub = len(publicas.Delegacion.unique())


print(n_delegacion_pub, n_niveles_pub)
print('delegacion',publicas.Delegacion.unique())
print('niveles',publicas.nivel.unique())
publicas


36 14
delegacion ['DELEGACION IZTACALCO' 'DELEGACION COYOACAN' 'DELEGACION TLAHUAC'
 'DELEGACION IZTAPALAPA' 'DELEGACION VENUSTIANO CARRANZA'
 'DELEGACION CUAUHTEMOC' 'DELEGACION TLALPAN' 'DELEGACION MILPA ALTA'
 'DELEGACION MIGUEL HIDALGO' 'DELEGACION ALVARO OBREGON'
 'DELEGACION CUAJIMALPA DE MORELOS' 'DELEGACION XOCHIMILCO' nan
 'DELEGACION GUSTAVO A. MADERO' 'DELEGACION BENITO JUAREZ'
 'BATALLONES ROJOS NO 205 LA ALBARRADA' 'AZCAPOTZALCO'
 'SALVADOR DIAZ MIRON' 'SELENE' 'ALVARO OBREGON'
 'AV DEL TRABAJO NO 258 MORELOS' 'DELEGACION ALVARO OBREGÓN' 'SANTA ROSA'
 'CERRADAS JUAN ESCUTIA' 'MEXICO DF.' 'PRIMARIA GENERAL'
 'DELEGACION AZCAPOTZALCO OBREGON' 'ADOLFO LOPEZ MATEOS' 'EL ROSARIO'
 'TELESECUNDARIA 89' 'EMILIO CARRANZA NO 39 ALBERT.' 'SAN JOSE'
 'SAN JUAN DE ARAGON' '553 Y 504 SECC. VECINAL DOS' '-UNIDAD HA'
 'SANTA CRUZ MEYEHUALCO']
niveles ['INTERVENCION' 'PRIMARIA' 'PREESCOLAR' 'ESCUELA' 'SECUNDARIA'
 'TELESECUNDARIA' 'LACTANTE,' 'SEC.' 'BRINDAR' 'SUPERVISAR' 'ATENDER'
 'LICE

nivel    nivel_2      nivel_3  ...      Ciudad Unnamed: 12 Unnamed: 13
0     INTERVENCION  TEMPRANA,  PREESCOLAR,  ...  MEXICO DF.         NaN         NaN
1         PRIMARIA    GENERAL          NaN  ...  MEXICO DF.         NaN         NaN
2       PREESCOLAR    GENERAL          NaN  ...  MEXICO DF.         NaN         NaN
3       PREESCOLAR    GENERAL          CON  ...  MEXICO DF.         NaN         NaN
4         PRIMARIA    GENERAL          NaN  ...  MEXICO DF.         NaN         NaN
...            ...        ...          ...  ...         ...         ...         ...
2378      PRIMARIA    GENERAL          NaN  ...  MEXICO DF.         NaN         NaN
2379    SECUNDARIA    GENERAL          NaN  ...         NaN         NaN         NaN
2380    PREESCOLAR    GENERAL          NaN  ...  MEXICO DF.         NaN         NaN
2381    SECUNDARIA    GENERAL          NaN  ...  MEXICO DF.         NaN         NaN
2382      PRIMARIA    GENERAL          NaN  ...         NaN         NaN         NaN

[2383 rows x 14 columns]

In [64]:
niveles_conjun = pd.merge(publicas, fidegar)
# niveles_conjun

## Index the nombres of each dataset

#### Merege and drop duplicates
Here we end up multiplying the datasets to get 3049 rows, which isn't bad. I dropped the duplicates and still got 3049 rows, this seems promising. The original fidegar has 2696 and the origianl escuelas publicas has 2383, i.e. fidegar has ~300 more schools listed.

> *I'm going to keep this set without duplicates and seet how that helps us.*

In [67]:
#cerate index of nombre and merge there
pub_nombre_idx = publicas.set_index('nombre')
fide_nombre_idx = fidegar.set_index('nombre')
# display(pub_nivel_idx, fide_nivel_idx)

conjun_nombre_idx = pd.merge(pub_nombre_idx, fide_nombre_idx,
                      left_index = True, right_index = True)
conjun_nombre_idx

nivel_x  nivel_2  ... ESCUELAS EN PROCESO JURÍDICO     nivel_y
nombre                                 ...                                         
10 DE ABRIL       PREESCOLAR  GENERAL  ...                          NaN  PREESCOLAR
14 DE JULIO         PRIMARIA  GENERAL  ...                          NaN    PRIMARIA
15 DE MAYO          PRIMARIA  GENERAL  ...                          NaN    PRIMARIA
15 DE SEPTIEMBRE    PRIMARIA  GENERAL  ...                          NaN    PRIMARIA
18 DE MARZO       PREESCOLAR  GENERAL  ...                          NaN    PRIMARIA
...                      ...      ...  ...                          ...         ...
YURI A. GAGARIN   SECUNDARIA  GENERAL  ...                          NaN  SECUNDARIA
ZACATENCO         PREESCOLAR  GENERAL  ...                          NaN  PREESCOLAR
ZAMBIA              PRIMARIA  GENERAL  ...                          NaN    PRIMARIA
ZIMBABWE            PRIMARIA  GENERAL  ...                          NaN    PRIMARIA
ZOFAI             PREESCOLAR  GENERAL  ...                          NaN  PREESCOLAR

[3049 rows x 23 columns]

In [81]:
# conjun_nombre_idx.to_excel(r'C:\Users\LW\Desktop\cdmx-escuelas_merged.xlsx', index = False, header=True)
# conjun_nombre_idx.to_csv(r'C:\Users\LW\Documents\Project Research\cdmx-escuelas_merged.csv', index = False, header=True)

conjun_nombre_idx.to_excel("C:\\Users\\LW\\Documents\\Project Research\\escuelas.xlsx")  

# with pd.ExcelWriter('cdmx-escuelas_merged.xlsx',
#                     mode='a') as writer:  
#     conjun_nombre_idx.to_excel(writer, sheet_name='conjun_nombre_idx')
    
    # conjun_nombre_idx.to_excel('cdmx-escuelas_merged.csv')


In [68]:
# conjun_nombre_idx = conjun_nombre_idx.drop(columns=['NIVEL', 'NOMBRE', 'CCT 1', 'CCT 2', 'CCT 3', 'CCT 4', 'MATRÍCULA', 'ALCALDÍA', 'ESCUELAS EN PROCESO JURÍDICO', 'nombre_y'])
conjun_nombre_idx = conjun_nombre_idx.drop_duplicates()
conjun_nombre_idx

nivel_x  nivel_2  ... ESCUELAS EN PROCESO JURÍDICO     nivel_y
nombre                                 ...                                         
10 DE ABRIL       PREESCOLAR  GENERAL  ...                          NaN  PREESCOLAR
14 DE JULIO         PRIMARIA  GENERAL  ...                          NaN    PRIMARIA
15 DE MAYO          PRIMARIA  GENERAL  ...                          NaN    PRIMARIA
15 DE SEPTIEMBRE    PRIMARIA  GENERAL  ...                          NaN    PRIMARIA
18 DE MARZO       PREESCOLAR  GENERAL  ...                          NaN    PRIMARIA
...                      ...      ...  ...                          ...         ...
YURI A. GAGARIN   SECUNDARIA  GENERAL  ...                          NaN  SECUNDARIA
ZACATENCO         PREESCOLAR  GENERAL  ...                          NaN  PREESCOLAR
ZAMBIA              PRIMARIA  GENERAL  ...                          NaN    PRIMARIA
ZIMBABWE            PRIMARIA  GENERAL  ...                          NaN    PRIMARIA
ZOFAI             PREESCOLAR  GENERAL  ...                          NaN  PREESCOLAR

[3049 rows x 23 columns]

## Index the niveles of each dataset

#### Merege and drop duplicates
Here we end up multiplying the datasets to get ~2220000 rows, then I dropped the duplicates and got only 2140 rows. The original fidegar has 2696 and the origianl escuelas publicas has 2383, i.e. fidegar has ~300 more schools listed.

In [54]:
#cerate index of nivel and merge there
pub_nivel_idx = publicas.set_index('nivel')
fide_nivel_idx = fidegar.set_index('nivel')
# display(pub_nivel_idx, fide_nivel_idx)

conjun_idx = pd.merge(pub_nivel_idx, fide_nivel_idx,
                      left_index = True, right_index = True)
conjun_idx


nivel_2  ...                      nombre_y
nivel                ...                              
PREESCOLAR  GENERAL  ...        RICARDO ZAMORA BUITRON
PREESCOLAR  GENERAL  ...            REPUBLICA DE KENIA
PREESCOLAR  GENERAL  ...            MAESTRO JOSE CALVO
PREESCOLAR  GENERAL  ...                ALFONSO QUIROZ
PREESCOLAR  GENERAL  ...  MA. EVELIA MONTERRUBIO SAENZ
...             ...  ...                           ...
SECUNDARIA  GENERAL  ...                     HUETZALIN
SECUNDARIA  GENERAL  ...                       OCELOTL
SECUNDARIA  GENERAL  ...             TOMAS ALVA EDISON
SECUNDARIA  GENERAL  ...               TLAMACHTILCALLI
SECUNDARIA  GENERAL  ...     GUILLERMO PRIETO PRADILLO

[2209225 rows x 23 columns]

In [58]:
conjun_idx = conjun_idx.drop(columns=['NIVEL', 'NOMBRE', 'CCT 1', 'CCT 2', 'CCT 3', 'CCT 4', 'MATRÍCULA', 'ALCALDÍA', 'ESCUELAS EN PROCESO JURÍDICO', 'nombre_y'])
conjun_idx = conjun_idx.drop_duplicates()
conjun_idx

nivel_2 nivel_3  ... Unnamed: 12 Unnamed: 13
nivel                        ...                        
PREESCOLAR  GENERAL     NaN  ...         NaN         NaN
PREESCOLAR  GENERAL     CON  ...         NaN         NaN
PREESCOLAR  GENERAL     CON  ...         NaN         NaN
PREESCOLAR  GENERAL     NaN  ...         NaN         NaN
PREESCOLAR  GENERAL     NaN  ...         NaN         NaN
...             ...     ...  ...         ...         ...
SECUNDARIA  GENERAL     NaN  ...         NaN         NaN
SECUNDARIA  GENERAL     NaN  ...         NaN         NaN
SECUNDARIA  GENERAL     NaN  ...         NaN         NaN
SECUNDARIA  GENERAL     NaN  ...         NaN         NaN
SECUNDARIA  GENERAL     NaN  ...         NaN         NaN

[2140 rows x 13 columns]

## Drop words to merge easier

here I try to play around with the stings in the columns to see if we could get matches ... it didn't exactly work.

In [ ]:
publicas['Delegacion'] = publicas['Delegacion'].str.strip('DELEGACION')

publicas

nivel    nivel_2      nivel_3  ...      Ciudad Unnamed: 12 Unnamed: 13
0     INTERVENCION  TEMPRANA,  PREESCOLAR,  ...  MEXICO DF.         NaN         NaN
1         PRIMARIA    GENERAL          NaN  ...  MEXICO DF.         NaN         NaN
2       PREESCOLAR    GENERAL          NaN  ...  MEXICO DF.         NaN         NaN
3       PREESCOLAR    GENERAL          CON  ...  MEXICO DF.         NaN         NaN
4         PRIMARIA    GENERAL          NaN  ...  MEXICO DF.         NaN         NaN
...            ...        ...          ...  ...         ...         ...         ...
2378      PRIMARIA    GENERAL          NaN  ...  MEXICO DF.         NaN         NaN
2379    SECUNDARIA    GENERAL          NaN  ...         NaN         NaN         NaN
2380    PREESCOLAR    GENERAL          NaN  ...  MEXICO DF.         NaN         NaN
2381    SECUNDARIA    GENERAL          NaN  ...  MEXICO DF.         NaN         NaN
2382      PRIMARIA    GENERAL          NaN  ...         NaN         NaN         NaN

[2383 rows x 14 columns]

In [ ]:
def clean_text(text):
  text = text.remove(r"DELEGACION", r'', text)
  text = ' '.join(text.split())
  return text

publicas['delegaciones'] = publicas['Delegacion'].apply(lambda x: clean_text(x))


In [ ]:
drop_wrd = ['DELEGACION']
strings = publicas.Delegacion[:]
strings = ''.join((filter(lambda i: i not in drop_wrd, strings)))
print(strings)

TypeError: ignored

In [ ]:
publicas.delegaciones = []

for i in publicas.Delegacion:
  delegaciones = str(publicas.Delegacion)
  words = delegaciones.split()
  if i in words == "DELEGACION":
    drop = words.remove('DELEGACION')
    redo = ''.join(drop)
    publicas.delegaciones.append(redo)

publicas


nivel    nivel_2      nivel_3  ...      Ciudad Unnamed: 12 Unnamed: 13
0     INTERVENCION  TEMPRANA,  PREESCOLAR,  ...  MEXICO DF.         NaN         NaN
1         PRIMARIA    GENERAL          NaN  ...  MEXICO DF.         NaN         NaN
2       PREESCOLAR    GENERAL          NaN  ...  MEXICO DF.         NaN         NaN
3       PREESCOLAR    GENERAL          CON  ...  MEXICO DF.         NaN         NaN
4         PRIMARIA    GENERAL          NaN  ...  MEXICO DF.         NaN         NaN
...            ...        ...          ...  ...         ...         ...         ...
2378      PRIMARIA    GENERAL          NaN  ...  MEXICO DF.         NaN         NaN
2379    SECUNDARIA    GENERAL          NaN  ...         NaN         NaN         NaN
2380    PREESCOLAR    GENERAL          NaN  ...  MEXICO DF.         NaN         NaN
2381    SECUNDARIA    GENERAL          NaN  ...  MEXICO DF.         NaN         NaN
2382      PRIMARIA    GENERAL          NaN  ...         NaN         NaN         NaN

[2383 rows x 14 columns]